### ISSUE TITLE
twisted.internet.test.test_process.ProcessTestsBuilder_SelectReactorTests.test_write hangs under Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8735]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8735|
|Type|enhancement|
|Created|2016-08-08 03:47:02Z|
|Branch|https://github.com/twisted/twisted/tree/8735-rodrigc-process-py3|

If I run the following command on Windows with Python 3.5:

```
python bin/trial twisted.internet.test.test_process.ProcessTestsBuilder_SelectReactorTests.test_write
```

the test hangs.

<details><summary>Searchable metadata</summary>

```
trac-id__8735 8735
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__Python_3_x Python-3.x
branch__8735_rodrigc_process_py3 8735-rodrigc-process-py3
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1470628022872359 1470628022872359
changetime__1470903604121729 1470903604121729
version__None None
owner__rodrigc rodrigc

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

https://github.com/twisted/twisted/pull/479

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @rodrigc|
|-|-|

Looks like the tests are passing and this is generally an improvement.  I had the one nitpick about MBCS encoding, but other than that, I think you can land at your leisure when you have a passing test run.  (I just merged trunk into the branch again to unblock merging, hopefully it will be a passing build when next you see it.)

Thanks for working on this!

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|GitHub set status to `closed`|
|-|-|

In changeset 343f4c0f0dc27c8129a2481c0ccb7a7487a58c6f
```
#!CommitTicketReference repository="" revision="343f4c0f0dc27c8129a2481c0ccb7a7487a58c6f"
Merge pull request #479 from twisted/8735-rodrigc-process-py3

Author: hawkowl, rodrigc
Reviewer: glyph
Fixes: #8735

Fix process related tests on Windows Python 3
```
